service/kvmconfig/v6/key/key.go

```diff
@@ -124,18 +128,54 @@ func MasterHostPathVolumeDir(clusterID string, vmNumber string) string {
 }
 
 // MemoryQuantity returns a resource.Quantity that represents the memory to be used by the nodes.
-// It adds the memory from the node definition parameter to the additional memory defined by
-// defaultWorkerMemory.
-func MemoryQuantity(n v1alpha1.KVMConfigSpecKVMNode) (resource.Quantity, error) {
-	q, err := resource.ParseQuantity(n.Memory)
-	if err != nil {
-		return resource.Quantity{}, microerror.Maskf(err, "creating Memory quantity from node definition")
-	}
-	additionalMemory, err := resource.ParseQuantity(defaultWorkerMemory)
-	if err != nil {
-		return resource.Quantity{}, microerror.Maskf(err, "creating Memory quantity from addtional memory")
+// It adds the memory from the node definition parameter to the additional memory calculated on the node role
+func MemoryQuantity(n v1alpha1.KVMConfigSpecKVMNode, role string) (resource.Quantity, error) {
+	var q resource.Quantity
+	var err error
+	if role == MasterID {
+		q, err = resource.ParseQuantity(n.Memory)
+		if err != nil {
+			return resource.Quantity{}, microerror.Maskf(err, "creating Memory quantity from node definition")
+		}
+		additionalMemory, err := resource.ParseQuantity(baseMasterMemoryOverhead)
```

Comment:
I would replace it with MustParse. If this value isn't parseable kvm-operator is basically unusable so it's fair to panic.

Reply:
done